the-collab-lab/tcl-7-smart-shopping-list
# Mm tr connect to firestore

## Description
In this PR, components needed to read and write from the Firestore database and changes in the app should be able to be reflected in the database and vice versa. Therefore, two components with each of these abilities were created: a **GetList** component which displays the grocery list of items and an **AddItem** component that allows items to be added to the list.

### The Beginnings

Before explaining the approach to creating the components, it must be pointed out that the **firebase.js** file had already been configured to the Firestore database which had also been set up prior. For further info, please check [here](https://firebase.google.com/docs/web/setup). In addition, we added the **FirestoreProvider** component (where we passed in the Firebase object as a prop) to allow it to communicate with Documents and Collections in the database. For further info, please read this [documentation](https://github.com/green-arrow/react-firestore) in regard to react-firestore which allowed our React components to fetch data from the database.

### **GetList: Retrieving Grocery List of Items**

In this component, the **FirestoreDocument** component was used by providing the respective path. This sets up a listener that updates when the database does and thereby, allows interaction with the database in real-time.

### **AddItem: Adding/Inserting Items**

In this component, we chose to create a simple form where the user can input the item by name. Upon form submission, the data will be sent to be updated in the Firestore database.

## Related Issue

Closes issue #2 

## Acceptance Criteria


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|   ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

There was nothing.

### After

![Screen Shot 2020-05-08 at 11 19 45 PM](https://user-images.githubusercontent.com/40173387/81462707-7e5cac00-9182-11ea-9b0c-369c4dbdd556.png)



## Testing Steps / QA Criteria

- Open the terminal and enter **git pull origin MM-TR-connect-to-firestore** and then, **git checkout MM-TR-connect-to-firestore**. 
- Once you've checked out of that branch, use npm install to install all your dependencies and then, npm start to open the project in your browser.
- When the app launches, place in an item of your choosing and see the updates in the database as shown [here](https://console.firebase.google.com/u/0/project/tcl-7-smart-shopping-list/database/firestore/data~2Fitems~2F6fd5C4jE5g4kz6Z9oULT).
